### PR TITLE
Added SBR Rails

### PR DESCRIPTION
**Pull request by jeroenrnl**
Issue #240
This change adds
- SBR rails. For now only SBR16
- The bearing block that goes with it (SBR16UU). I made a split in `bearing_blocks` to support a new kind of bearing block. I tried to reuse some code, but due to the fact that the letters used to indicate measurements are vastly different between SBR and SCS - well and the fact that the blocks are quite different as well - this was only possible in a very limited way.
- An 'open' linear bearing (LM16UUOP) that is used by the bearing block, I made some changes in the `linear_bearing[s].scad` to support this.
- A 28mm circlip that has a wider opening than then normal circlip.
(circlip_28iw)

preview

It should be fairly easy to add the other types of rails, bearing blocks and linear bearings, but I first wanted to hear whether my current approach is the right one.
I used [this PDF](https://cnc4you.co.uk/resources/SBRxxUU.pdf), as well as a set of SBR16 rails I had on hand and several other online drawings I could find.

The 28mm 'wide' open circlip is something I could not find any documentation on, so the measurements are mostly guessed, based on the one that is in the SBR16UU I have. The name '*circlip_28iw*' is something I made up.

### CODE REVIEW

File: vitamins/bearing_block.scad [hunk 1]
```diff
@@ -169,3 +170,147 @@ module scs_bearing_block_assembly(type, part_thickness, screw_type, nut_type) {
...
+
+    translate([0,-part_thickness,0])
+        sbr_bearing_block_hole_positions(type)
+            rotate([0,180,0])
+                screw(screw_type, screw_length);
+}
+
+/* Generic bearing block functions */
+
+module bearing_block_hole_positions(type, y) { //! Place children at hole positions
```
Comment (nophead) [inline vitamins/bearing_block.scad:303]: This doesn't seem to be generic. All the scs blocks now have their screws in the sbr positions, which is wrong.

**Review by nophead — COMMENTED** [review]
Looks good apart from the SCS screw positions.

File: vitamins/bearing_block.scad (hunk 1 above)
Comment (jeroenrnl) [reply vitamins/bearing_block.scad:303]: Weird I didn't catch that, as I did check the results on the SCS blocks... Looking into it.

Comment (jeroenrnl) [reply vitamins/bearing_block.scad]: > This doesn't seem to be generic. All the scs blocks now have their screws in the sbr positions, which is wrong.

It should be fixed now. I removed the generic function as after the change the functions had very little in common now.

**Comment by jeroenrnl** [comment]
I also rebased the branch on master so it has the changes you made in the past few days incorporated.

**Comment by nophead** [comment]
I took the liberty of deriving the open angle from the open width. I can't see how they can not be exactly related. On the data sheets the numbers vary a bit but that seems to just be rounding as only one can be an integer.

**Pull request merged**